src/commands/mod.rs

```diff
@@ -991,38 +991,22 @@ impl DaemonControl {
     ) -> ListTransactionsResult {
         let mut db_conn = self.db.connection();
         let txids = db_conn.list_txids(start, end, limit);
-        let transactions = txids
-            .iter()
-            .filter_map(|txid| {
-                // TODO: batch those calls to the Bitcoin backend
-                // so it can in turn optimize its queries.
-                self.bitcoin
-                    .wallet_transaction(txid)
-                    .map(|(tx, block)| TransactionInfo {
-                        tx,
-                        height: block.map(|b| b.height),
-                        time: block.map(|b| b.time),
-                    })
-            })
+        let transactions = db_conn
+            .list_wallet_transactions(&txids)
+            .into_iter()
+            .map(|(tx, height, time)| TransactionInfo { tx, height, time })
             .collect();
         ListTransactionsResult { transactions }
```

Comment:
i think we can call directly `list_transaction()` here?
```rust
        let txids = self.db.connection().list_txids(start, end, limit);
        self.list_transactions(&txids)
```

Reply:
Ah yes, you're right. I'll change that.